Flickr/Sources/FlickrAppVCs/SearchTabCoordinator.swift

```diff
@@ -55,13 +56,38 @@ extension SearchTabCoordinator: SearchScreenViewControllerDelegate {
 }
 
 extension SearchTabCoordinator: SearchResultsViewControllerDelegate {
-    func didSelectImage(url: URL, title: String, imageTitle: String) {
-        let photoViewerVC = PhotoViewerVC(url: url, imageTitle: imageTitle)
+    func didSelectImage(url: URL, title: String, imageTitle: String, imageId: String) {
+        let photoViewerVC = PhotoViewerVC(url: url, imageTitle: imageTitle, imageId: imageId, imageData: nil)
         photoViewerVC.title = title
+        photoViewerVC.photoViewerDelegate = self
         rootNavigationController?.pushViewController(photoViewerVC, animated: true)
     }
 }
 
+extension SearchTabCoordinator: PhotoViewerViewControllerDelegate {
+    func pushToFavorites(imageData: Data, id: String, title: String) {
+        var favorites = FavoriteImagesArray(array: [])
+        let newArray = FileManagerCoordinator.retrieveData() ?? []
+        favorites.array = newArray
+        let image = FavoriteImageData(imageId: id, imageData: imageData, imageTitle: title)
+        favorites.array.append(image)
+        favoritesArray = favorites
+        FileManagerCoordinator.storeData(favoritesArray)
+    }
+
+    func popFromFavorites(id: String) {
```

Comment:
Can the naming of the functions be improved here? The names do ne exactly convey the purpose of the functions.